src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CaseReferenceValidator.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import com.google.common.base.Strings;
+import io.vavr.control.Validation;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.Optional;
+import javax.annotation.Nonnull;
+
+import static io.vavr.control.Validation.invalid;
+import static java.lang.String.format;
+
+class CaseReferenceValidator {
+    private static final String ATTACH_TO_CASE_REFERENCE = "attachToCaseReference";
+
+    @Nonnull
+    Validation<String, String> validate(CaseDetails theCase) {
+        return Optional.ofNullable(theCase)
```

Comment:
As we're only making specific checks here, we could as well do sth like this:
```
String caseRef = getCaseRef(theCase); // alternatively, this could be optional, and then you could map, etc.

if (caseRef == null) {
    return invalid("No case reference supplied");
} else {
    return validateCaseRef(caseRef)
}
```

My main point is that we can extract some of the complexity into a separate method, which also names that bit.

Not very strong about it, as it's not crucial what approach we end up with. It's a bottom level thing.

Reply:
looks much cleaner and easier to read what @lgonczar has suggested